Test.hs

```diff
@@ -22,72 +21,71 @@ main = do
   putStrLn "VHDL:"
   testVHDL
   putStrLn "\n------All Tests Passed------"
-  where
 
 testShowInstance :: IO ()
-testShowInstance = case a == (show b) of
-  True -> putStrLn "-Pangraph show instance passed."
-  False -> error $ "!Show instance failed test: " ++ show b ++ "!=" ++ a
-  where
-    a = fst showInstanceTestCase
-    b = snd showInstanceTestCase
-    showInstanceTestCase :: (String, P.Pangraph)
-    showInstanceTestCase = ("Pangraph [Node [Attribute (\"id\",\"0\")],Node [Attribute (\"id\",\"1\")]] [Edge [Attribute (\"source\",\"0\"),Attribute (\"target\",\"1\")]]",
-     P.makePangraph [P.makeNode [P.makeAttribute ("id","0")],P.makeNode [P.makeAttribute ("id","1")]] [P.makeEdge [P.makeAttribute ("source","0"),P.makeAttribute ("target","1")]])
+testShowInstance = do
+  literal <-fmap (head . lines) $ readFile "test/show-string.txt"
+  let graph = fromMaybe
+                (error "Sample graph failed to build") $
+                P.makePangraph
+                  sampleVertices [
+                  P.makeEdge
+                    [("source","0"), ("target","1")]
+                    (head sampleVertices, sampleVertices !! 1)
+                  ]
+      sampleVertices = [P.makeVertex "0" [("id","0")]
+                       ,P.makeVertex "1" [ ("id","1")]]
+  if literal == show (graph :: P.Pangraph)
+    then putStrLn "-Instance passed"
+    else error
+      $ "!Show instance failed test: \""
+      ++ literal
+      ++ "\"\nSample(above) != Graph(below)\n\""
+      ++ show graph
+        ++ "\""
+
     -- Without escapes:
-    -- Pangraph [Node [Attribute ("id","0")],Node [Attribute ("id","1")]] [Edge [Attribute ("source","0"),Attribute ("target","1")]]
+    -- makePangraph [makeVertex "0" [("id","0")],makeVertex "1" [("id","1")]] [makeEdge [("source","0"),("target","1")] (makeVertex "0" [("id","0")],makeVertex "1" [("id","1")])]
 
 testGraphML :: IO ()
 testGraphML = do
-  file <- fmap pack $ readFile path
-  case (parser (head GraphML_P.template) file) of
-    Left x -> error (show x)
-    Right y -> if y == graphs
-      then putStrLn "-Parse Passed"
-      else error $ "!Test failed\n" ++ show y
-  where
-    path :: FilePath
-    path = "examples/graphs/small.graphml"
-    parser :: Parser
-    parser = GraphML_P.parseTemplateToPangraph
-    graphs :: P.Pangraph
-    graphs =
-      P.makePangraph
-        [P.makeNode [P.makeAttribute ("id","n0")],
-        P.makeNode [P.makeAttribute ("id","n1")],
-        P.makeNode [P.makeAttribute ("id","n2")]]
-        [P.makeEdge
-          [P.makeAttribute ("source","n0"),
-          P.makeAttribute ("target","n2")]]
+  file <- BS.readFile "examples/graphs/small.graphml"
+  let graph  = fromMaybe (error "Sample graph failed to build") graph'
+      graph' = P.makePangraph sampleVertices
+                [P.makeEdge
+                  [("source","n0"),("target","n2")]
+                  (head sampleVertices, sampleVertices !! 2)]
+      sampleVertices = [
```

Comment:
Same for `sampleVertices` -- this works thanks to laziness, but let's help the reader and write things in the evaluation order, whenever possible.